tests/Integration/Metadata/CustomTaxonomyTermArchiveTest.php

```diff
@@ -76,7 +78,8 @@ public function test_metadata_is_correctly_constructed_for_custom_taxonomy_term_
 		// Create the structured data for that term archive. The term archive
 		// metadata doesn't use the post data, but the construction method
 		// requires it for now.
-		$metadata        = new Metadata( $parsely );
+		$metadata = new Metadata( $parsely );
+		/** @var array<string, mixed> $structured_data */
```

Comment:
Instead of giving type here can we return this type in `construct_metadata` function?

Reply:
The `construct_metadata` function already returns a PHPStan type which is more specific, and thus would be considered better at least theoretically.

```
	/**
	 * Creates Parse.ly metadata object from post metadata.
	 *
	 * @param WP_Post $post object.
	 *
	 * @return Metadata_Attributes
	 */
	public function construct_metadata( WP_Post $post ) {
```

The above type seems to be incompatible with the array type that is expected in another function a bit below. So this is why we are casting it to array here. Maybe it would be possible to fix this by modifying `construct_metadata` or the receiving function, but this was out of scope for the PR.

Maybe this could be the object of a future iteration on PHPStan-related improvements.

Reply:
Sidenote: In general, PHPStan types seem to have some side-effects in my experience, and they can be the reason for the `Out of X possible param types, only X % actually have it. Add more param types to get over X %` errors mentioned previously.

So in the future, we can look at all the data and see how to proceed regarding that.